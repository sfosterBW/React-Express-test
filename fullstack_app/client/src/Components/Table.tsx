//Libraries
import React, { Component, ReactNode } from 'react'

interface Props{
  rows: ReactNode
  title: string
}

export default class Table extends Component<Props> {

  render(): ReactNode{
    return(
      <div id="fruit-table">
        <h2>{this.props.title}</h2>
        <table>
          <thead>
            <tr>
              <th>Fruit</th>
              <th>Best?</th>
              <th>Edit</th>
            </tr>
          </thead>
          <tbody>{this.props.rows}</tbody>
        </table>
      </div>
    )
  }
}
