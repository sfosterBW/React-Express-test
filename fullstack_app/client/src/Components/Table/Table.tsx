import React, { FC, useState } from 'react'
import { IFruit } from '../../utils/interfaces'
import Row from '../Row/Row'
import styles from './Table.module.css'

interface Props {
  fruits: IFruit[]
}

const Table: FC<Props> = ({ fruits }) => {

  const type: boolean | undefined = undefined
  const [filter, setFilter] = useState<boolean|undefined>(type)

  const displayFruitList = () => {
    const filteredFruit = fruits
      .filter((i) => filter === i.best || filter === undefined)
    if(filteredFruit.length > 0) {
      return filteredFruit
        .map((i) => <Row key={`${i._id}${i.name}`} fruit={i} />)
    } else {
      return <p>Nothing here...</p>
    }
  }

  return (
    <section className={styles.wrapper}>
      <div className={styles.filters}>
        <button className={styles.filter} onClick={() => setFilter(undefined)}>All</button>
        <button className={styles.filter} onClick={() => setFilter(true)}>Best</button>
        <button className={styles.filter} onClick={() => setFilter(false)}>Not best</button>
      </div>
      <div className={styles.devices}>
        {displayFruitList()}
      </div>
    </section>
  )
}

export default Table
