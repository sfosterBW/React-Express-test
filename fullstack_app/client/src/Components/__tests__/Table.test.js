
import React from 'react'
import { mount, render, shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import Table from '../Table'

describe('the Table component', () => {
  it('shallow renders', () => {
    const table = mount(<Table />)
    expect(table).toBeDefined()
  })


  it('renders the same as last time', () => {
    const tree = renderer
      .create(<Table />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
