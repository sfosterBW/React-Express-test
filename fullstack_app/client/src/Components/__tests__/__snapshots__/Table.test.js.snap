// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Table component renders the same as last time 1`] = `
<div
  className="fruit-table"
>
  <h2 />
  <table>
    <thead>
      <tr>
        <th>
          ID
        </th>
        <th>
          Fruit
        </th>
        <th>
          Best?
        </th>
        <th>
          Edit
        </th>
        <th>
          Update
        </th>
        <th>
          Remove
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
